plugins/decoder.rst

```diff
@@ -1,4 +1,34 @@
 Decoder Plugin
 ==============
 
-TODO
+Decoder Plugin decodes the body of the response with filters coming from the `Transfer-Encoding` or `Content-Encoding`
+headers::
+
+    use Http\Discovery\HttpClientDiscovery;
+    use Http\Client\Plugin\PluginClient;
+    use Http\Client\Plugin\DecoderPlugin;
+
+    $decoderPlugin = new DecoderPlugin();
+
+    $pluginClient = new PluginClient(
+        HttpClientDiscovery::find(),
+        [$decoderPlugin]
+    );
+
+Actually it can decodes 4 different filters:
+
+ * chunked: Decode a stream with a chunked encoding (no size in the `Content-Length` header of the response)
+ * compress: Decode a stream encoded with the compress filter according to `RFC 1950`_
+ * Deflate: Decode a stream encoded with the inflate filter according to `RFC 1951`_
+ * gzip: Decode a stream encoded with the gzip filter according to `RFC 1952`_
+
+You can also use the decoder plugin to only decodes the `Transfer-Encoding` header and not the `Content-Encoding` one
+by setting the first parameter of the constructor to false::
+
+    $decoderPlugin = new DecoderPlugin(false);
```

Comment:
actually makes me think we should refactor this constructor to use options and the OptionsResolver. even if there will be no other options (which you never know), things would be more readable: `new DecoderPlugin(['content_encoding' => false]);`